src/Cli/Microsoft.TemplateEngine.Cli/PostActionProcessors/AddJsonPropertyPostActionProcessor.cs

```diff
@@ -158,22 +185,20 @@ internal class AddJsonPropertyPostActionProcessor : PostActionProcessorBase
 
             do
             {
-                List<string> filesInDir = fileSystem.EnumerateFileSystemEntries(directory, matchPattern, SearchOption.AllDirectories).ToList();
-                List<string> matches = new();
-
-                matches = secondaryFilter == null ? filesInDir : filesInDir.Where(x => secondaryFilter(x)).ToList();
+                Reporter.Verbose.WriteLine(string.Format(LocalizableStrings.PostAction_ModifyJson_Verbose_AttemptingToFindJsonFile, matchPattern, directory));
+                string[] filesInDir = fileSystem.EnumerateFileSystemEntries(directory, matchPattern, SearchOption.AllDirectories).ToArray();
 
-                if (matches.Count > 0)
+                if (filesInDir.Length > 0)
                 {
-                    return matches;
+                    return filesInDir;
                 }
 
                 directory = Path.GetPathRoot(directory) != directory ? Directory.GetParent(directory)?.FullName : null;
                 numberOfUpLevels++;
             }
-            while (directory != null && numberOfUpLevels <= maxAllowedAboveDirectories);
+            while (directory != null && numberOfUpLevels <= 1);
```

Comment:
It'd be good to have a better walk-up logic but I couldn't think of anything that would work for all/most cases.

1. Walking up to first project found is problematic for templates adding project as we would never walk-up.
2. Walking up to first sln found can lead to walking up to root if no sln exists
3. Walking up to repo root is hard because we have nothing to understand where to stop (is it git folder? is it something else?).